JavaScript/hoisting.md

```diff
@@ -95,36 +101,88 @@ console.log(‘I like very much’);
 is: ‘like = undefined;’ so it gives you an error ‘like is not a function’. If, however, instead of on line 1 you call it on the
 last line of the code, then during the execution phase, JavaScripts assigns ‘like’ its value which in this case is a function 
 (remember that assignments happen during the execution phase) and then when it hits the last line it can call it.
 
-##Recap
-Let me give you one more example which I found on w3schools that demonstrates hoisting rather nicely:
+## Recap
+Let me give you two different example that demonstrates hoisting rather nicely:          
+Example 1:
+```javascript
+function family(){
+    function person() {
+        return "John";
+    }
+    return person();
+    function person() {
+        return "Tom";
+    }
+}
+console.log(family());
+```
 
+Example 2:
 ```javascript
-console.log(x); //this produces ‘undefined’
-x = 5;
-console.log(x); //this produces 5
-//some code here
-var x;
+function money(){
+    return coins();
+    var coins = function() {
+        return 3;
+    };
+    var coins = function() {
+        return 8;
+    };
+}
+console.log(money());
 ```
 
-When we execute the code, we first go through the creation phase and then the execution phase. During the creation phase, 
-variable ‘x’ is put into memory and assigned the value undefined. Now that the creation phase is over, the execution phase begins.
-JavaScript goes through the code line by line and sees the first console.log. At this point, x = undefined so it
-console.logs ‘undefined’. Then it sees the assignment of 5 to x. Remember, assignments happen during the execution phase.
-So x is now 5. Then it sees another console.log and console.logs 5 because at this point 5 has already been assigned to x.
+So, the answer to Ex. 1 would be ```"Tom" ``` and the answer to Ex. 2 would be ```TypeError: coins is not a function ```
 
-I hope this document has shed some light on what hoisting is and how it works. Thank you for reading.
+Explanation for Ex. 1 - So we know that we are dealing with function declarations in this example, therefore we can say that the functions and their body is hoisted, so after the interpreter is finished with the code, it runs in this order.
+```javascript
+//**Simulated processing sequence for Ex. 1**
+function family(){
+    //define person once
+    function person() {
+        return "John";
+    }
+    //redefine person as "Tom"
+    function person() {
+        return "Tom";
+    }
+    return person(); //"Tom"
+}
+console.log(family());
+```
 
-##Related
+Explanation for Ex. 2 - Here we can see that function expressions are used, so no functions are hoisted. However if there were no hoisting in this example whatsoever, the error would be ```TypeError: coins is not a defined ```. What occurs is variable hoisting, thus coins is declared up front but it's value is undefined. Everything else runs to order.
```

Comment:
I would change it to "thus `coins` is declared up front but its value is `undefined`. Everything else runs in order."